text/0223-cicd-observability-OTEP.md

```diff
@@ -0,0 +1,75 @@
+
+Some tools have started adopting OpenTelemetry, which is an important step in creating standardization. A good example is [Jenkins](https://github.com/jenkinsci/jenkins), a popular CI OSS project, which offers the [Jenkins OpenTelemetry plugin](https://plugins.jenkins.io/opentelemetry/) for emitting telemetry data in order to:
+1. Visualize jobs and pipelines executions as distributed traces
+2. Visualize Jenkins and pipeline health indicators
+3. Troubleshoot Jenkins performances with distributed tracing of HTTPs requests
+
+Building CI/CD observability involves four stages: Collect → Store → Visualize → Alert. OpenTelemetry provides a unified way for the first step, namely collecting and ingesting the telemetry data in an open and uniform manner. 
+
+If you are a CI/CD tool builder, the specification and instrumentation will enable you to properly structure your telemetry, package and emit it over OTLP. OpenTelemetry specification will determine which data to collect, the semantic convention of the data, and how different signal types can be correlated based on that, to support downstream analytics of that data by various tools.
+
+If you are an end user looking to gain observability over your pipelines, you will be able to collect OpenTelemetry-formatted telemetry using the OpenTelemetry Collector, ingest, process and then export to a rich ecosystem of observability analytics backend tools, independent of your CI/CD tools in use.     
+
+Here are some examples of potential resulting observability visualization over popular backend tools such as Jaeger, Grafana and OpenSearch:
+
+Monitoring Jenkins metrics for nodes, queues, jobs and executors with Grafana dashboard:
+![Monitoring Jenkins metrics for nodes, queues, jobs and executors with Grafana dashboard](https://dytvr9ot2sszz.cloudfront.net/wp-content/uploads/2022/05/image6.png "Monitoring Jenkins metrics for nodes, queues, jobs and executors with Grafana dashboard")
+
+Jenkins pipeline run visualized as a trace in the Timeline View in Jaeger UI:
+![Jenkins pipeline run visualized as a trace in the Timeline View in Jaeger UI](https://dytvr9ot2sszz.cloudfront.net/wp-content/uploads/2022/05/image9.png)
+
+OpenSearch dashboard for monitoring Jenkins pipelines:
+![OpenSearch dashboard for monitoring Jenkins pipelines](https://dytvr9ot2sszz.cloudfront.net/wp-content/uploads/2022/05/image7.png)
+
+For more examples, see [this article](https://logz.io/learn/cicd-observability-jenkins/) on CI/CD observability using currently available open source tools.
```

Comment:
Could we add more articles and presentations here? We have been a bunch in the last three years.

* [Improve your software delivery with CI/CD observability and OpenTelemetry](https://www.elastic.co/virtual-events/ci-cd-observability-and-opentelemetry)
* [DevOpsWorld 2021 - Embracing Observability in Jenkins with OpenTelemetry](https://www.youtube.com/watch?v=3XzVOxvNpGM)
* [DevOpsWorld 2021 - Who Observes the Watchers? An Observability Journey](https://www.cloudbees.com/videos/who-observes-the-watchers-an-observability-journey?wvideo=wv11m4uazu)
* [Embracing Observability in CI/CD with OpenTelemetry](https://www.slideshare.net/cyrille.leclerc/embracing-observability-in-cicd-with-opentelemetry)
* [FOSDEM 2022 - OpenTelemetry and CI/CD](https://archive.fosdem.org/2022/schedule/event/opentelemetry_and_ci_cd/)
* [cdCon Austin 2022 - Making your CI/CD Pipelines Speaking in Tongues with OpenTelemetry](https://www.youtube.com/watch?v=1jDLNNe_TEM)